config/config.go

```diff
@@ -84,35 +86,30 @@ func (cfg *Config) Validate() error {
 		}
 	}
 
-	// Check that all enabled extensions in the service are configured.
-	if err := cfg.validateServiceExtensions(); err != nil {
+	return cfg.validateService()
+}
+
+func (cfg *Config) validateService() error {
+	// Validate Telemetry
+	if err := cfg.Service.Telemetry.validate(); err != nil {
 		return err
 	}
 
-	// Check that all pipelines have at least one receiver and one exporter, and they reference
-	// only configured components.
-	return cfg.validateServicePipelines()
-}
-
-func (cfg *Config) validateServiceExtensions() error {
-	// Validate extensions.
+	// Check that all enabled extensions in the service are configured.
 	for _, ref := range cfg.Service.Extensions {
 		// Check that the name referenced in the Service extensions exists in the top-level extensions.
 		if cfg.Extensions[ref] == nil {
 			return fmt.Errorf("service references extension %q which does not exist", ref)
 		}
 	}
 
-	return nil
-}
-
-func (cfg *Config) validateServicePipelines() error {
 	// Must have at least one pipeline.
 	if len(cfg.Service.Pipelines) == 0 {
 		return errMissingServicePipelines
 	}
 
-	// Validate pipelines.
+	// Check that all pipelines have at least one receiver and one exporter, and they reference
+	// only configured components.
 	for _, pipeline := range cfg.Service.Pipelines {
```

Comment:
These changes seem unrelated to the PR topic. Maybe do in a separate PR.

Reply:
I restructure the validate to be able to add the validate for the encoding around the same place with the other validation for the service.